src/model/writer.js

```diff
@@ -794,6 +794,10 @@ export default class Writer {
 	 * {@link module:engine/model/markercollection~Marker marker class description} to learn about the difference between
 	 * markers managed by operations and not-managed by operations.
 	 *
+	 * The `options.affectData` parameter, which defaults to `true`, allows you to create marker, that does not affects the data model.
+	 * It impacts the performance, because changes in markers with set `options.affectData:false` do not cause firing the
+	 * {@link module:engine/model/document~Document#event:change:data `change:data`} event.
```

Comment:
I would change this sentence. I am not sure if firing `change:data` event has an impact on the performance. I would focus on:
 - `affectData = true` fires `change:data`, `affectData = false` fires `change` event,
 - `affectData` should be true when the maker change changes data return by `editor.getData`.